Add check if `/t:` needs to be add on windows

elmarburke opened the pull request:
From Node v10.8.0 on, node-gyp comes in v3.7.0 which will prepend it as a target. Added a check which version is used to not always add the prefix.

(check also https://stackoverflow.com/questions/51526377/the-target-tnative-metrics-does-not-exist-in-the-project/52086917 and https://discuss.newrelic.com/t/newrelic-native-metrics-does-not-install-on-windows-10-from-corrupt-vcxproj/54625/3 which looks quite similar)

NatalieWolfe commented on lib/pre-build.js:
```diff
@@ -18,6 +18,7 @@ var zlib = require('zlib')
 
 var CPU_COUNT = os.cpus().length
 var IS_WIN = process.platform === 'win32'
+var HAS_NEW_NODE_GYP_ARGS = process.env.versions.node >= '10.8.0'
```
The `versions` property is directly on `process` (no `.env`). https://nodejs.org/api/process.html#process_process_versions

Doing a string comparison is also not good, this expression evaluates to true: `'9.0.0' >= '10.8.0'`. It is probably fine to move `semver` from a dev-dep to a normal dependency and use that. This module should always be installed along with `newrelic`, which has `semver` as a dep already, thus it wont increase the install overhead.

elmarburke replied:
Good one - I've added `semver` to the deps and check with it. Also, I've swapped the logical connection to make reading simpler.

NatalieWolfe commented on lib/pre-build.js:
```diff
@@ -186,7 +187,7 @@ function execGyp(args, cb) {
 
 function build(target, rebuild, cb) {
-  if (IS_WIN) {
+  if (IS_WIN && HAS_NEW_NODE_GYP_ARGS) {
```
I believe this logic should be negated (`&& !HAS_NEW_NODE_GYP_ARGS`)? On the newer versions of gyp we do not want to prepend `/t:`, but older versioned still need it.

NatalieWolfe commented:
Thanks for the contribution! It's great to see gyp resolve the target issue on Windows. If you make the corrections from the review I'll make sure this gets merged and released ASAP.

elmarburke commented:
Thanks for the fast review, I've addressed your points in the last two commits.

I've also added the semver licence to the unit test, not sure if this is as expected.

![](https://media.giphy.com/media/ppFaBYPO8paUg/giphy.gif)

NatalieWolfe commented:
Awesome, thanks! That license check is indeed expected, I just forgot about it. Our legal team likes to make sure we stay on top of the licenses we use. :)

NatalieWolfe commented:
I've scheduled a release for this on Monday. Thanks again for the contribution!
